p4rt_app/sonic/state_verification.cc

```diff
@@ -200,5 +245,47 @@ std::vector<std::string> VerifyAppStateDbAndAppDbEntries(
       ReadAllEntriesFromRedisTable(app_state_db, "AppStateDb"));
 }
 
+std::vector<std::string> VerifyP4rtTableWithCacheTableEntries(
+    TableAdapter& app_db,
+    const absl::flat_hash_map<gutil::TableEntryKey, p4::v1::TableEntry>&
+        table_entry_cache,
+    const pdpi::IrP4Info& ir_p4_info, bool translate_port_ids,
+    const boost::bimap<std::string, std::string>& port_translation_map) {
+  std::vector<std::string> failures;
+
+  // Translate the Table cache into IR entries for comparison.
+  std::vector<pdpi::IrTableEntry> ir_entries;
+  int p4rt_translation_failures = 0;
+  for (const auto& [_, pi_table_entry] : table_entry_cache) {
+    auto ir_table_entry = TranslatePiTableEntryForOrchAgent(
+        pi_table_entry, ir_p4_info, translate_port_ids, port_translation_map,
+        /*translate_key_only=*/false);
+    if (ir_table_entry.ok()) {
+      ir_entries.push_back(*std::move(ir_table_entry));
+    } else {
+      p4rt_translation_failures++;
+    }
+  }
+  if (p4rt_translation_failures > 0) {
+    failures.push_back(absl::StrCat("Failed to translate ",
+                                    p4rt_translation_failures,
+                                    " entries from the table entry cache."));
+  }
+
+  RedisTable redis_db_entries = ReadAllEntriesFromRedisTable(app_db, "AppDb");
+
+  // Remove any entries for ACL table definitions.
+  absl::erase_if(redis_db_entries.entries, [](const auto& iter) {
+    return (absl::StartsWith(iter.first, "ACL_TABLE_DEFINITION_TABLE:") || absl::StartsWith(iter.first, "TABLES_DEFINITION_TABLE:"));
```

Comment:
This has not been formatted correctly, column width cannot exceed 80 chars in a line.